Simple.OData.Client.V4.Adapter/ResponseReader.cs

```diff
@@ -250,23 +250,35 @@ private ODataMediaAnnotations CreateAnnotations(ODataStreamReferenceValue value)
 
         private object GetPropertyValue(object value)
         {
-            if (value is ODataResource)
+            if (value is ODataResource resource)
             {
-                return ((ODataResource) value).Properties.ToDictionary(
-                    x => x.Name, x => GetPropertyValue(x.Value));
+                return resource.Properties.ToDictionary(x => x.Name, x => GetPropertyValue(x.Value));
             }
-            else if (value is ODataCollectionValue)
+            else if (value is ODataCollectionValue collectionValue)
             {
-                return ((ODataCollectionValue) value).Items
-                    .Select(GetPropertyValue).ToList();
+                return collectionValue.Items.Select(GetPropertyValue).ToList();
             }
-            else if (value is ODataEnumValue)
+            else if (value is ODataEnumValue enumValue)
             {
-                return ((ODataEnumValue) value).Value;
+                return enumValue.Value;
             }
-            else if (value is ODataUntypedValue)
+            else if (value is ODataUntypedValue untypedValue)
             {
-                return ((ODataUntypedValue) value).RawValue;
+                var result = untypedValue.RawValue;
+                if (!string.IsNullOrEmpty(result))
+                {
+                    // Remove extra quoting as has been read as a string
+                    // Don't just replace \" in case we have embedded quotes
+                    if (result.StartsWith("\""))
+                    {
+                        result = result.Substring(1);
+                    }
+                    if (result.EndsWith("\""))
```

Comment:
If you are caring about embedded quotes and the surrounding quotes seem to be optional, I think those two ifs should be combined to be prepared for results like `endswithquote\"` or `\"halfquoted`:

``` csharp
if (result.StartsWith("\"") && result.EndsWith("\""))
{
  result = result.Substring(1, result.Length - 1);
}
```

Reply:
Yes, that sounds sensible